pkg/component/controller/apiserver.go

```diff
@@ -230,7 +234,7 @@ func (a *APIServer) Ready() error {
 		TLSClientConfig: tlsConfig,
 	}
 	client := &http.Client{Transport: tr}
-	resp, err := client.Get(fmt.Sprintf("https://localhost:%d/readyz?verbose", a.ClusterConfig.Spec.API.Port))
+	resp, err := client.Get(fmt.Sprintf("https://%s:%d/readyz?verbose", a.ClusterConfig.Spec.API.Address, a.ClusterConfig.Spec.API.Port))
```

Comment:
Ideally, this would use `net.JoinHostPort`. I think it'll break for IPv6 addresses otherwise.